examples/timeseries/timeseries_classification_from_scratch.py

```diff
@@ -0,0 +1,247 @@
+plt.legend(loc="best")
+plt.show()
+plt.close()
+
+"""
+## Standardizing the data
+
+Our time series are already in a single length (176). However, their values are
+usually in various ranges. This is not ideal for a neural network;
+in general we should seek to make the input values normalized. Here, we will
+standardize values to have a mean equal to zero and a standard deviation equal to one.
+We perform this step for each train and test set.
+"""
+
+std_ = x_train.std(axis=1, keepdims=True)
+std_[std_ == 0] = 1.0
+x_train = (x_train - x_train.mean(axis=1, keepdims=True)) / std_
+
+std_ = x_test.std(axis=1, keepdims=True)
+std_[std_ == 0] = 1.0
+x_test = (x_test - x_test.mean(axis=1, keepdims=True)) / std_
+
+"""
+Note that the time series data used here are univariate, meaning we only have one channel
+per time series example.
+We will therefore transform the time series into a multivariate one with one channel
+using a simple reshaping via numpy.
+This will allow us to construct a model that is easily applicable to multivariate time
+series.
+"""
+
+x_train = x_train.reshape((x_train.shape[0], x_train.shape[1], 1))
+x_test = x_test.reshape((x_test.shape[0], x_test.shape[1], 1))
+
+"""
+Finally, in order to use `sparse_categorical_crossentropy`, we will have to count
+the number of classes beforehand.
+"""
+
+nb_classes = len(np.unique(y_train))
+
+"""
+Now we shuffle the training set because we will be using the `validation_split` option
+later when training.
+"""
+
+idx = np.random.permutation(len(x_train))
+x_train = x_train[idx]
+y_train = y_train[idx]
+
+"""
+Standardize the labels to positive integers.
+"""
+
+y_train[y_train == -1] = 0
+y_test[y_test == -1] = 0
+
+"""
+## Build a model
+
+We build a Fully Convolutional Neural Network originally proposed in
+[this paper](https://arxiv.org/abs/1611.06455).
+The implementation is based on the TF2.0 version provided
+[here](https://github.com/hfawaz/dl-4-tsc/).
+
+"""
+
+
+def make_model(input_shape, num_classes):
+    input_layer = keras.layers.Input(input_shape)
+
+    # This is the first 1D convolutional block.
+    # First a linear convolution is applied by sliding a certain number of filters
+    # on the input time series to transform it into a multivariate time series,
+    # whose number of dimensions is equal to the number of filters used here.
+    conv1 = keras.layers.Conv1D(filters=128, kernel_size=8, padding="same")(input_layer)
+    # Following the convolution, we apply a batchnormalization in order to
+    # help the network converge quickly while adding an implicit regularization.
+    conv1 = keras.layers.BatchNormalization()(conv1)
+    # Finally we add the relu activation function to inject some non-linearity
+    # to the convolutions' output.
+    conv1 = keras.layers.Activation(activation="relu")(conv1)
+
+    # A second convolutional block is applied.
+    conv2 = keras.layers.Conv1D(filters=256, kernel_size=5, padding="same")(conv1)
```

Comment:
Indeed, perhaps KerasTuner would give some insights for choosing the best hyperparameters. 
I haven't tried it though. I simply used the same model/hyperparameters published [here](https://ieeexplore.ieee.org/abstract/document/7966039).
Nevertheless it would be interesting to see what KerasTuner would give on this dataset. 
Should I apply it "offline" or include it in this tutorial ?

Reply:
Please use it offline to find the best model parameters. You can then add a comment in the code explaining how you found these parameters.